contribs/gmf/src/query/gridComponent.js

```diff
@@ -478,18 +478,26 @@ QueryGridController.prototype.getMergedSource_ = function (source, mergedSources
     return null;
   }
 
+  /** @type {boolean} */
+  let newRequest = true;
+
   /** @type {import('ngeo/statemanager/WfsPermalink.js').QueryResultSource} */
   let mergeSource;
   if (mergeSourceId in mergedSources) {
     mergeSource = mergedSources[mergeSourceId];
+    if (source.requestPartners) {
+      newRequest = !source.requestPartners.some((label) => mergeSource.mergeComposants.includes(label));
+    }
+    mergeSource.mergeComposants.push(source.label);
   } else {
     mergeSource = {
       features: [],
       id: mergeSourceId,
       label: mergeSourceId,
-      limit: this.maxResults,
+      limit: 0,
```

Comment:
Are you sure of this change ?

Reply:
The number of results is not always the given maxResults. If the merged sources come from different request each source has the maxResults which have to added. Also (see bug https://jira.camptocamp.com/browse/GSGMF-1457) the number of results obtained not always corresponds to maximum number of results allowed. Therefore the limit is updated later.

Reply:
ok, thanks